File: String/Java/Form_the_Largest_Number.java
import java.util.Arrays;
import java.util.Comparator;

class Solution {
    public String findLargest(int[] arr) {
        String strArr[] = new String[arr.length];
        for (int i = 0; i < arr.length; i++)
            strArr[i] = Integer.toString(arr[i]);
        Arrays.sort(strArr, new Comparator<String>() {
            @Override
            public int compare(String a, String b) {
                return (b + a).compareTo(a + b);
            }
        });
        StringBuilder ans = new StringBuilder();
        for (String i : strArr) {
            ans.append(i);
        }
        return ans.charAt(0) == '0' ? "0" : ans.toString();
    }
}

public class Form_the_Largest_Number {
    public static void main(String[] args) {
        Solution sol = new Solution();
        System.out.println(sol.findLargest(new int[] { 3, 30, 34, 5, 9 }));// 9534330
        System.out.println(sol.findLargest(new int[] { 54, 546, 548, 60 }));// 6054854654
        System.out.println(sol.findLargest(new int[] { 3, 4, 6, 5, 9 }));// 96543
    }
}

File: String/C++/Form_the_Largest_Number.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;
class Solution
{
public:
    string findLargest(vector<int> &arr)
    {
        vector<string> strArr;
        for (int i : arr)
            strArr.push_back(to_string(i));
        sort(strArr.begin(), strArr.end(), [](const string &a, const string &b)
             { return (b + a) < (a + b); });
        string ans = "";
        for (string i : strArr)
            ans += i;
        return ans[0] == '0' ? "0" : ans;
    }
};
int main()
{
    Solution sol;
    vector<int> arr = {54, 546, 548, 60};
    cout << sol.findLargest(arr) << "\n";
    arr = {3, 30, 34, 5, 9};
    cout << sol.findLargest(arr) << "\n";
    arr = {3, 4, 6, 5, 9};
    cout << sol.findLargest(arr) << "\n";
    return 0;
}
